graspologic/plot/plot.py

```diff
@@ -671,6 +674,248 @@ def pairplot(
+        colors = sns.color_palette(label_palette, n_components)
+        label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+    # must give labels AND palettes
+    if (labels is not None) and (cluster_palette is None or label_palette is None):
+        msg = "if give labels array must give palette dicts or string desciptors"
+        raise ValueError(msg)
+    elif labels is None and (cluster_palette is not None or label_palette is not None):
+        msg = "must give labels if using palettes for labels and clusters"
+        raise ValueError(msg)
+    # This takes care of default case, when no palettes or labels given
+    elif labels is None and cluster_palette is None and label_palette is None:
+        colors = sns.color_palette("Set1", n_components)
+        cluster_palette = dict(zip(np.unique(lab_names), colors))
+        label_palette = dict(zip(np.unique(lab_names), colors))
+
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        # handle the case where only given two dims
+        if X.shape[1] == 2:
+            pairplot, axes = plt.subplots(1, 1, figsize=(12, 12))
+            _plot_ellipse(
+                data,
+                X,
+                0,
+                1,
+                means,
+                covariances,
+                axes,
+                label_palette,
+                cluster_palette,
+                alpha=alpha,
+            )
+            # formatting
+            axes.set_ylabel("Dim " + str(0))
+            axes.set_xlabel("Dim " + str(1))
+            axes.label_outer()
+            axes.spines["right"].set_visible(False)
+            axes.spines["top"].set_visible(False)
+            handles, labels = [], []
+            handles_, labels_ = axes.get_legend_handles_labels()
+            handles += handles_
+            labels += labels_
+            pairplot.legend(
+                handles[: means.shape[0] + 1],
+                labels[: means.shape[0] + 1],
+                loc="upper right",
+            )
+            if title:
+                pairplot.suptitle(title)
+            else:
+                pairplot.suptitle(
+                    "Fit a Gaussian mixture using "
+                    + str(means.shape[0])
+                    + " components"
+                )
+            return pairplot
+        # for the case with more than 2 dims
+        pairplot, axes = plt.subplots(X.shape[1], X.shape[1], figsize=(12, 12))
+        for k in range(X.shape[1]):
+            for j in range(X.shape[1]):
+                if k == j:
+                    # take care of the distplot on diagonal
+                    for t, lab in zip([i for i in range(X.shape[1])], label_palette):
+                        sns.distplot(
+                            X[Y_ == t, k],
+                            kde=True,
+                            ax=axes[k, k],
+                            color=label_palette[lab],
+                        )
```

Comment:
I think the user will need more access to these params potentially - for instance, KDEs sometimes break when the data in a class is a point-mass (KDE cannot be computed here). Maybe a `distplot_kws` argument?

Reply:
also, distplot is deprecated in seaborn 0.11. Have a look at the replacements (I think you'll want `histplot`). Unfortunately we still say we support <0.11 so not sure what to do here. related to #594

Reply:
@bdpedigo I have added a kde argument. Hopefully this will allow for better access.

Reply:
I introduced a KDE arg https://github.com/microsoft/graspologic/pull/541/commits/df314e2811d30e834ed5213cbf301723763f4efa . I also changed to histplot https://github.com/microsoft/graspologic/pull/541/commits/cd57c0be377de22a93e9ba8be3ab048bee94784a